include/envoy/upstream/upstream.h

```diff
@@ -62,6 +62,16 @@ class Host : virtual public HostDescription {
   createConnection(Event::Dispatcher& dispatcher,
                    const Network::ConnectionSocket::OptionsSharedPtr& options) const PURE;
 
+  /**
+   * Create a health check connection for this host.
+   * @param dispatcher supplies the owning dispatcher.
+   * @param options supplies the socket options that will be set on the new connection.
```

Comment:
I think socket options can be dropped here for now, as it's not used. We can add it back later as needed.

Reply:
Ok got it. I have updated it.